sdks/go/pkg/beam/core/runtime/harness/logging.go

```diff
@@ -61,7 +61,7 @@ func (l *logger) Log(ctx context.Context, sev log.Severity, calldepth int, msg s
 		Severity:  convertSeverity(sev),
 		Message:   msg,
 	}
-	if _, file, line, ok := runtime.Caller(calldepth); ok {
+	if _, file, line, ok := runtime.Caller(calldepth + 1); ok {
```

Comment:
Correct! The logging in this spot here was never accounting for itself, which meant higher levels needed to +1 more to get it to work properly. This meant that anyone who was replacing the base logger (with some other logging package for example) either needed to replicate the mistake (that is, no +1), or be incorrect.

Reply:
Ah, I think I see. I misunderstood the description of the error as being that the call depth was always 1 off. But if I'm understanding correctly now, it's only 1 off when replacing the base logger, right?

If I'm understanding correctly: the previous implementation was getting the right result in an incorrect way when using the base logger, so if it got replaced by a logger that did things correctly (adding 1 to the calldepth per stack frame), the end result would be wrong since replacement logger wouldn't work around the mistake like the base logger was.